codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java

```diff
@@ -251,31 +251,47 @@ public static boolean isContentLengthSet(HttpMessage m) {
     }
 
     /**
-     * Returns {@code true} if and only if the specified message contains the
-     * {@code "Expect: 100-continue"} header.
+     * Returns {@code true} if and only if the specified message contains an expect header and the only expectation
+     * present is the 100-continue expectation. Note that this method returns {@code false} if the expect header is
+     * not valid for the message (e.g., the message is a response, or the version on the message is HTTP/1.0).
+     *
+     * @param message the message
+     * @return {@code true} if and only if the expectation 100-continue is present and it is the only expectation
+     * present
      */
     public static boolean is100ContinueExpected(HttpMessage message) {
-        // Expect: 100-continue is for requests only.
-        if (!(message instanceof HttpRequest)) {
+        if (!isExpectHeaderValid(message)) {
             return false;
         }
 
-        // It works only on HTTP/1.1 or later.
-        if (message.protocolVersion().compareTo(HttpVersion.HTTP_1_1) < 0) {
-            return false;
-        }
+        final String expectValue = message.headers().get(HttpHeaderNames.EXPECT);
+        // unquoted tokens in the expect header are case-insensitive, thus 100-continue is case insensitive
+        return HttpHeaderValues.CONTINUE.toString().equalsIgnoreCase(expectValue);
+    }
 
-        // In most cases, there will be one or zero 'Expect' header.
-        CharSequence value = message.headers().get(HttpHeaderNames.EXPECT);
-        if (value == null) {
+    /**
+     * Returns {@code true} if the specified message contains an expect header specifying an expectation that is not
+     * supported. Note that this method returns {@code false} if the expect header is not valid for the message
+     * (e.g., the message is a response, or the version on the message is HTTP/1.0).
+     *
+     * @param message the message
+     * @return {@code true} if and only if an expectation is present that is not supported
+     */
+    static boolean isUnsupportedExpectation(HttpMessage message) {
+        if (!isExpectHeaderValid(message)) {
             return false;
         }
-        if (HttpHeaderValues.CONTINUE.contentEqualsIgnoreCase(value)) {
-            return true;
-        }
 
-        // Multiple 'Expect' headers.  Search through them.
-        return message.headers().contains(HttpHeaderNames.EXPECT, HttpHeaderValues.CONTINUE, true);
+        final String expectValue = message.headers().get(HttpHeaderNames.EXPECT);
+        // TODO: do not allow expectation extensions that do not conform to the grammar?
```

Comment:
can you add a link to the RFC which defines the grammar? [section 5.1.1](https://tools.ietf.org/html/rfc7231#section-5.1.1) seems pretty limited on this. Also this method in not intended to be extensible and it seems like `newContinueResponse` would have to be the extension point ... so maybe its OK as is (just reject everything but `HttpHeaderValues.CONTINUE`)